Orbitersdk/samples/ProjectApollo/src_sys/apolloguidance.cpp

```diff
@@ -567,6 +582,9 @@ void ApolloGuidance::LoadState(FILEHANDLE scn)
 		papiReadScenario_bool(line, "PROGALARM", ProgAlarm);
 		papiReadScenario_bool(line, "GIMBALLOCKALARM", GimbalLockAlarm);
 	}
+
+	if (!scalerLoaded)
```

Comment:
Wouldn't this also work by checking if vagc.Scaler == 0? This would avoid introducing a new variable.

Reply:
One thing what I learned during my career that if I don't want to cause terrible headache for myself or for any other poor bastard then I shouldn't use 0 as a missing value :) Additionally as I mentioned in my previous response, there will be other new scen paramteres like CDU counter registers which I plane to read in here as well, and there getting a 0 as loaded value is much more likely than in case of the scaler, so with this trick we could mess up the saved simulation state. And apart from that: why does the introduction of a new local variable like this means any problem? It takes max. 4 bytes from the stack in a code which runs only once in a session so no measurable impact on performance, and it also makes the intention of the programmer much more clearer.

Reply:
I got the upstream yaAGC to run on NASSP today. Still need to test that everything works as it should but looks to be working so far.

Reply:
> I got the upstream yaAGC to run on NASSP today. Still need to test that everything works as it should but looks to be working so far.

That's really great, thanks! Am I supposed to use the sync_agc (seem to me the same as fix_agc)? During the weekend I'll build up something similar to this PR but on top of sync_agc and submit it. Is Do you concur?

Reply:
There is still some work to be done. I didn't manage to do a landing with the new AGC yet (partially because I don't know how to fly them LEM yet) so there might still be some issues.
Then yaAGS also needs another look as it currently depends on yaAGC to build. It would be easier to separate those before introducing new changes.

What we can do is once this stuff is mainlined: adapt your PR to work with it and then test it in a feature branch. If everything is working as it should you can submit those changes to VirtualAGC to get them upstreamed. That way we won't diverge again. I'll make another commit today with some final fixes so you can get started on porting it over. The rest of the changes to come will be minor so shouldn't present problems with getting your work integrated.